thijsvanloef/palworld-server-docker
# Add Daily Reboot Cronjob (K8s)

Requires the k8s deployment to be in the **palworld** namespace.



Currently, it is set to run at 9:30am UTC daily.

I'm newer to configuring such things, and will work to improve/cleanup a little bit. :)

## Context 

* This launches a cronjob to run a shell script (defined in cm.yaml), which sends a broadcast notifying players that the server is saving. After 30 seconds, it does a formal backup using the established 'exec pod/podnamehere backup' command in the repo. After 30 seconds later, it sends a broadcast notifying players that the server will reboot in 60 seconds; 30 seconds; then each second for the last 10 seconds. It the runs the "Shutdown 1" command via rcon-cli, waits 30 additional seconds, then redeploys the application.

## Choices

* I prefer the K8s cluster schedule and perform a cronjob so I can keep track of it and also rollback if needed.
* This made the most sense to me, as it does not intrude or alter the main image itself.
* I wanted to learn more about how CronJobs work in Kubernetes.

## Test instructions

1. Copy the folder and its contents
2. Edit the cronjob.yaml schedule to '* * * * *' so it schedules it immediately.
3. Log into server
4. Apply the yaml files, specifically applying cronjob.yaml last.
5. Wait for the broadcasts, also monitor the logs of the job created using `kubectl -n palworld logs job/jobnamehere`
6. Server should disconnect once timer reaches 0; redeployment should be under way.
7. Server should come back up once redeployment is finished.

## Checklist before requesting a review
